pkg/controller/master/node/cpu_manager_controller.go

```diff
@@ -152,9 +183,51 @@ func (h *cpuManagerNodeHandler) OnJobChanged(_ string, job *batchv1.Job) (*batch
 		return nil, h.updateCPUManagerStatus(node, job, CPUManagerFailedStatus)
 	}
 
+	// someone delete the cpu manager job
+	if job.DeletionTimestamp != nil {
+		logrus.WithField("job_name", job.Name).Warn("CPU Manager job job was deleted before it finished")
+		return nil, h.updateCPUManagerStatus(node, job, CPUManagerFailedStatus)
+	}
+
 	return job, nil
 }
 
+func (h *cpuManagerNodeHandler) OnPodChanged(_ string, pod *corev1.Pod) (*corev1.Pod, error) {
+	if pod == nil || pod.Labels == nil || pod.DeletionTimestamp != nil {
+		return pod, nil
+	}
+	jobName, found := pod.Labels[batchv1.JobNameLabel]
+	exitCode := int32(0)
+	if len(pod.Status.ContainerStatuses) > 0 && pod.Status.ContainerStatuses[0].State.Terminated != nil {
+		exitCode = pod.Status.ContainerStatuses[0].State.Terminated.ExitCode
+	}
+	if found && strings.HasPrefix(jobName, CPUManagerJobPrefix) && exitCode > 0 {
+		// Update exit code to cpu manager job
+		job, err := h.jobCache.Get(h.namespace, jobName)
+		if err != nil {
```

Comment:
For `IsNotFound` error, may consider to log message and return nil.

Reply:
IMO, there is no need to specifically handle the IsNotFound case in Line206, if any error raise here, we should throw error. Job is expected to be there if pod still exist.

I'll modify the code to
```
		if err != nil {
			logrus.WithField("job_name", jobName).WithError(err).Error("Failed to get cpu manager job from cache")
			return nil, err
		}
```

Reply:
Whe a Job is deleted, the related Pod is also deleted, both meet the `.DeletionTimestamp != nil`;  but there is no assurance that the Pod will be gone first and the Job will be gone then. (I did not check the k8s source code yet);  hence, 
```
pod.DeletionTimestamp != nil && IsNotFound(job)
```
is better to be escaped, otherwise, if above assumption is true, here it may stuck reporting error.

Reply:
Roger that, will handle the IsNotFound case, thanks.

Reply:
gentle ping @w13915984028 , I've addressed the comment in the latest commit, also rebased to master branch. PTAL, thanks.